api/src/main/java/io/smallrye/reactive/messaging/annotations/Blocking.java

```diff
@@ -0,0 +1,47 @@
+package io.smallrye.reactive.messaging.annotations;
+
+import static java.lang.annotation.ElementType.METHOD;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies that a particular method performs blocking operations,
+ * and as such should be executed on separate worker.
+ *
+ * When possible, the initial threads execution context is _restored_
```

Comment:
I think the term "initial threads" could be a bit confusing for some non-English speakers as the apostrophe is missing it is close to "the initiating thread's context is propogated". It noun phrase would be better as
"When possible, the context on the thread used to execute the method is restored to the context present prior to the method being executed"    (if you want that meaning). 
Or "When possible, the context present on the worker thread when the blocking method is run is set to match the context used to run non-blocking methods." (if you want that meaning)....and apologies for "bike shedding".

Reply:
What about changing "initial" to "triggering"?

Reply:
"When supported, the caller execution context is captured and restored after the execution of the blocking method. This means that the rest of the pipeline would run in the same context (same thread). Typically, this is the case if the caller is using a Vert.x Context (event loop or worker)."